project/_vars.sls

```diff
@@ -1,7 +1,9 @@
 {% set root_dir = "/var/www/" + pillar['project_name'] + "/" %}
 
-{% macro get_primary_ip(ifaces) -%}
-  {{ ifaces.get(salt['pillar.get']('primary_iface', 'eth0'), {}).get('inet', [{}])[0].get('address') }}
+{% macro get_primary_ip(host, ifaces) -%}
+  {# use `primary_iface` from pillar, then fall back to getting the iface from the default route #}
+  {%- set minion_iface = salt['pillar.get']('primary_iface', salt['mine.get'](host, 'network.default_route')[host][0]['interface']) -%}
```

Comment:
I think (?) that because we import _vars.sls `with context`, that setting a variable here will set it globally?  It looks like we don't really need a temporary variable here, so it might be safer not to use `{% set %}` here.